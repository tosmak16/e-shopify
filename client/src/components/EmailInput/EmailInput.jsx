import { useState, useEffect } from 'react';

import TextInput, { INPUT_TYPES } from '../TextInput/TextInput';
import styles from './EmailInput.scss';

const EmailInput = props => {
  const [email, setEmail] = useState({
    emailValue: '',
    emailError: ''
  });
  const { emailValue, emailError } = email;
  const { shouldShowCheckMark, updatedEmail, updateEmail, shouldShowErrorMessage } = props;

  const validateInput = value => {
    let errorMessage;
    if (!value.length) {
      errorMessage = 'Please enter your email';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(value)) {
      errorMessage = 'Please enter a valid email';
    } else {
      errorMessage = '';
    }
    return errorMessage;
  };

  const inputChange = event => {
    const { value } = event.target;
    const error = validateInput(value);
    setEmail({ emailValue: value, emailError: error });
    updateEmail({ value, error, isValid: error.length === 0 });
  };

  useEffect(() => {
    const { value, error } = updatedEmail;
    setEmail({ emailValue: value, emailError: error });
  }, [updatedEmail]);
  return (
    <div>
      <TextInput
        {...props}
        type={INPUT_TYPES.EMAIL}
        isValid={!emailError.length}
        onChange={inputChange}
        value={emailValue}
        inputDidChange={shouldShowCheckMark}
      />
      {emailError.length > 0 && shouldShowErrorMessage && (
        <span styleName="input-validation-message-label">{emailError}</span>
      )}
    </div>
  );
};

EmailInput.defaultProps = {
  shouldShowCheckMark: false,
  updatedEmail: {
    value: '',
    error: ''
  },
  shouldShowErrorMessage: true
};

EmailInput.propTypes = {
  updateEmail: PropTypes.func.isRequired,
  shouldShowCheckMark: PropTypes.bool,
  updatedEmail: PropTypes.objectOf(PropTypes.any),
  shouldShowErrorMessage: PropTypes.bool
};
export default CSSModules(EmailInput, styles, { allowMultiple: true });
