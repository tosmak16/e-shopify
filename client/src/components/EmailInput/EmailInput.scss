@import "~variables";

.input-validation-message-label {
    @include input-validation-text;
    font-family: $main-font;
    word-spacing: -4px;
    letter-spacing: -1px;

}