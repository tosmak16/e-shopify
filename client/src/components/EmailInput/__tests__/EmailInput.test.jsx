import EmailInput from '../EmailInput';

describe('EmailInput', () => {
	const initialProps = {
		id: 'email',
		name: 'email',
		updateEmail: jest.fn(),
	};
	it('renders', () => {
		const tree = shallow(
			<EmailInput
				{...initialProps}
			/>
		);
		expect(tree).toMatchSnapshot();
	});

	it('should call updateEmail function when inputChange method is called with valid email value', () => {
		const tree = shallow(
			<EmailInput
				{...initialProps}
			/>
		);

		const event = {
			target: {
				value: 'test@mobiliti.com'
			}
		};
		const textInputObject = tree.find('TextInput');
		textInputObject.props().onChange(event);
		expect(initialProps.updateEmail).toBeCalled();
	});
});
