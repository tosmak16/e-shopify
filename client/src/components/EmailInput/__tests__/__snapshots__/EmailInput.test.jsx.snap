// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput renders 1`] = `
<div>
  <TextInput
    className=""
    dark={false}
    disabled={false}
    errorMessage=""
    id="email"
    inputDidChange={false}
    isValid={true}
    label=""
    name="email"
    onBlur={null}
    onChange={[Function]}
    onChangePasswordVisibilityStatus={[Function]}
    onKeyDown={[Function]}
    placeholder=""
    shouldShowCheckMark={false}
    shouldShowErrorMessage={true}
    shouldShowVisibility={false}
    type="email"
    updateEmail={[MockFunction]}
    updatedEmail={
      Object {
        "error": "",
        "value": "",
      }
    }
    value=""
  />
</div>
`;
